### PR TITLE
Update links and documentation to fit new monorepo structure

### LINKED ISSUE #5439: Update the ./documentation to have correct monorepo guidance

### PR DESCRIPTION
**Pull request by sam-b-rose**
### WHY are these changes introduced?

Fixes #5439 

This PR addresses some doc updates related to
- the migration to a monorepo structure
- the rename of the repo from `polaris-react` to `polaris`

### CODE REVIEW

File: documentation/Tophatting documentation.md [hunk 1]
```diff
@@ -6,20 +6,24 @@ When components are updated in ways that impact how people use them, their `READ
...
 ## What should you 🎩?
 
 Check the location, appearance, and readability of the content you add to the style guide.
 
+<!-- To be replaced by monorepo method soon
+
 ## How to 🎩 documentation
 
-1. Check out the `master` branch from [`polaris-styleguide`](https://github.com/Shopify/polaris-styleguide)
+1. Check out the `main` branch from [`polaris-styleguide`](https://github.com/Shopify/polaris-styleguide)
```
Comment (sam-b-rose) [inline documentation/Tophatting documentation.md:23]: What do we think we should do with this section for now?

Related to the changes and recommendations [in our `CONTRIBUTING.md`](https://github.com/Shopify/polaris/pull/5523#discussion_r849817750)

File: documentation/guides/migrating-from-v3-to-v4.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ export function mountWithPolaris(node, context = {}) {
 
 ### AppProvider <a name="polaris-appprovider"></a>
 
-Polaris now supports multiple languages and ships with [many translations](https://github.com/Shopify/polaris-react/blob/main/locales) to support internationalization. The `i18n` prop is now required to specify the translations you wish to use.
+Polaris now supports multiple languages and ships with [many translations](https://github.com/Shopify/polaris/blob/main/locales) to support internationalization. The `i18n` prop is now required to specify the translations you wish to use.
```
Comment (aaronccasanova) [inline documentation/guides/migrating-from-v3-to-v4.md]:
```suggestion
Polaris now supports multiple languages and ships with [many translations](https://github.com/Shopify/polaris/blob/main/polaris-react/locales) to support internationalization. The `i18n` prop is now required to specify the translations you wish to use.
```

`/locales` is still located in `/polaris-react`

File: documentation/guides/migrating-from-v3-to-v4.md [hunk 3]
```diff
@@ -76,7 +76,7 @@ import translations from '@shopify/polaris/locales/en.json';
 <AppProvider i18n={translations}>
 ```
 
-If you use [`@shopify/react-i18n`](https://github.com/Shopify/quilt/tree/master/packages/react-i18n) and want to dynamically load translations based on a provided locale, see the [AppProvider README](https://github.com/Shopify/polaris-react/blob/main/src/components/AppProvider/README.md#using-translations) for more information.
+If you use [`@shopify/react-i18n`](https://github.com/Shopify/quilt/tree/main/packages/react-i18n) and want to dynamically load translations based on a provided locale, see the [AppProvider README](https://github.com/Shopify/polaris/blob/main/src/components/AppProvider/README.md#using-translations) for more information.
```
Comment (aaronccasanova) [inline documentation/guides/migrating-from-v3-to-v4.md]:
```suggestion
If you use [`@shopify/react-i18n`](https://github.com/Shopify/quilt/tree/main/packages/react-i18n) and want to dynamically load translations based on a provided locale, see the [AppProvider README](https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/AppProvider/README.md#using-translations) for more information.
```

**Review by aaronccasanova — APPROVED** [review]
Now that `@shopify/polaris` documentation lives at the root of the monorepo I wonder how we can make it more clear that pages like the migration guides, `Accessibility testing`, `Generating styles`, `Tophatting`, `Versioning and changelog`, etc. are specific to the `polaris-react` workspace. We can address this in follow up PRs but think it's worth having on our radar.

Updates look great though! Left a few small suggestions

**Pull request merged**